peripherals/adc/adc_v3.yaml

```diff
@@ -0,0 +1,246 @@
+    OVSS: [0, 11]
+    JOVSE:
+      Disabled: [0, "Injected oversampling disabled"]
+      Enabled: [1, "Injected oversampling enabled"]
+    ROVSE:
+      Disabled: [0, "Regular oversampling disabled"]
+      Enabled: [1, "Regular oversampling enabled"]
+  CFGR,JSQR:
+    EXTEN,JEXTEN:
+      Disabled: [0, "Trigger detection disabled"]
+      RisingEdge: [1, "Trigger detection on the rising edge"]
+      FallingEdge: [2, "Trigger detection on the falling edge"]
+      BothEdges: [3, "Trigger detection on both the rising and falling edges"]
+  CFGR:
+    EXTSEL:
+      TIM1_CC1: [0, "Timer 1 CC1 event"]
+      TIM1_CC2: [1, "Timer 1 CC2 event"]
+      TIM1_CC3: [2, "Timer 1 CC3 event"]
+      TIM2_CC2: [3, "Timer 2 CC2 event"]
+      TIM3_TRGO: [4, "Timer 3 TRGO event"]
+      TIM4_CC4: [5, "Timer 4 CC4 event"]
+      EXTI11: [6, "EXTI line 11"]
+      TIM8_TRGO: [7, "Timer 8 TRGO event"]
+      TIM8_TRGO2: [8, "Timer 8 TRGO2 event"]
+      TIM1_TRGO: [9, "Timer 1 TRGO event"]
+      TIM1_TRGO2: [10, "Timer 1 TRGO2 event"]
+      TIM2_TRGO: [11, "Timer 2 TRGO event"]
+      TIM4_TRGO: [12, "Timer 4 TRGO event"]
+      TIM6_TRGO: [13, "Timer 6 TRGO event"]
+      TIM15_TRGO: [14, "Timer 15 TRGO event"]
+      TIM3_CC4: [15, "Timer 3 CC4 event"]
+      HRTIM1_ADCTRG1: [16, "HRTIM1_ADCTRG1 event"]
+      HRTIM1_ADCTRG3: [17, "HRTIM1_ADCTRG3 event"]
+      LPTIM1_OUT: [18, "LPTIM1_OUT event"]
+      LPTIM2_OUT: [19, "LPTIM2_OUT event"]
+      LPTIM3_OUT: [20, "LPTIM3_OUT event"]
+  JSQR:
+    JEXTSEL:
+      TIM1_TRGO: [0, "Timer 1 TRGO event"]
+      TIM1_CC4: [1, "Timer 1 CC4 event"]
+      TIM2_TRGO: [2, "Timer 2 TRGO event"]
+      TIM2_CC1: [3, "Timer 2 CC1 event"]
+      TIM3_CC4: [4, "Timer 3 CC4 event"]
+      TIM4_TRGO: [5, "Timer 4 TRGO event"]
+      EXTI15: [6, "EXTI line 15"]
+      TIM8_CC4: [7, "Timer 8 CC4 event"]
+      TIM1_TRGO2: [8, "Timer 1 TRGO2 event"]
+      TIM8_TRGO: [9, "Timer 8 TRGO event"]
+      TIM8_TRGO2: [10, "Timer 8 TRGO2 event"]
+      TIM3_CC3: [11, "Timer 3 CC3 event"]
+      TIM3_TRGO: [12, "Timer 3 TRGO event"]
+      TIM3_CC1: [13, "Timer 3 CC1 event"]
+      TIM6_TRGO: [14, "Timer 6 TRGO event"]
+      TIM15_TRGO: [15, "Timer 15 TRGO event"]
+      HRTIM1_ADCTRG2: [16, "HRTIM1_ADCTRG2 event"]
+      HRTIM1_ADCTRG4: [17, "HRTIM1_ADCTRG4 event"]
+      LPTIM1_OUT: [18, "LPTIM1_OUT event"]
+      LPTIM2_OUT: [19, "LPTIM2_OUT event"]
+      LPTIM3_OUT: [20, "LPTIM3_OUT event"]
+  # Sample time
+  "SMPR?":
+    "SMP*":
+      Cycles1_5: [0, "1.5 ADC clock cycles"]
+      Cycles2_5: [1, "2.5 ADC clock cycles"]
+      Cycles8_5: [2, "8.5 ADC clock cycles"]
+      Cycles16_5: [3, "16.5 ADC clock cycles"]
+      Cycles32_5: [4, "32.5 ADC clock cycles"]
+      Cycles64_5: [5, "64.5 ADC clock cycles"]
+      Cycles387_5: [6, "387.5 ADC clock cycles"]
+      Cycles810_5: [7, "810.5 ADC clock cycles"]
+  # Channel preselection
+  PCSEL:
+    "PCSEL*":
```

Comment:
All the `PCSEL` bits are plain read-write so I think should have an enum for 0 (to un-preselect a channel) and not be in `_write`?

Reply:
I was making an assumption that the only valid usage is to preselect... However this isn't the place to restrict something that isn't restricted in the RM. Not even sure if my assumption is valid either!